src/NumericalAlgorithms/LinearOperators/Divergence.cpp

```diff
@@ -27,6 +27,22 @@ Scalar<DataVector> divergence(
     const Mesh<Dim>& mesh,
     const InverseJacobian<DataVector, Dim, Frame::Logical, DerivativeFrame>&
         inverse_jacobian) noexcept {
+  Scalar<DataVector> div_input{mesh.number_of_grid_points()};
+  divergence(make_not_null(&div_input), input, mesh, inverse_jacobian);
+  return div_input;
+}
+
+template <size_t Dim, typename DerivativeFrame>
+void divergence(
+    const gsl::not_null<Scalar<DataVector>*> div_input,
+    const tnsr::I<DataVector, Dim, DerivativeFrame>& input,
+    const Mesh<Dim>& mesh,
+    const InverseJacobian<DataVector, Dim, Frame::Logical, DerivativeFrame>&
+        inverse_jacobian) noexcept {
+  if (UNLIKELY(get(*div_input).size() != mesh.number_of_grid_points())) {
```

Comment:
Could this if be replaced with `destructive_resize_components`?

Reply:
yes, I squashed this in